### DISCUSSION TITLE
Clarification questions: SAFE Phase 2 simulation

### DISCUSSION THREAD

**Discussion by Rinus-Verdonschot-UB** [asreview/asreview#1881] [Q&A] [answered]
Dear all,

I have taken an annotated dataset (N=3878) and simulated it with the standard SAFE Phase 1 parameters (TF-IDF, NB, etc).
According to the simulation, if this was done manually I should have stopped around 30% (Fig 1).

![afbeelding](https://github.com/user-attachments/assets/5a8b18d0-13ac-4d9e-af88-2031071df05e)
**Figure 1**. _SAFE Phase 1_

Then I created a simulation which stopped at 30% (at N=1164) using --stop_if 1164 as parameter.
I exported this dataset to see the row numbers which had been included in the simulation.

These row numbers I included (yes I know pandas starts with 0 instead of 1) as priors in a Phase 2 simulation with sBERT/RF.

Then I got the following figure (Fig 2)

![afbeelding](https://github.com/user-attachments/assets/26e9e17f-8279-4e96-9f52-f47ea49db804)
**Figure 2**. _SAFE Phase 2_

Sorry for, once again, asking stupid questions, but I am curious as to the following

(_edited_)
(1) **Clarification**: the data represented in Figure 2 is the **remaining data** (i.e. data minus phase 1 priors) to be analyzed, right? So, this would be (3878 [total]-1164 [phase 1]) 2714 remaining items, of which Total of 1s minus prior 1s remains as "1". So, then in the first 2-3% I find about 80%+ of the remaining "1s" (so the non-prior 1s remaining after 30% of phase 1).

(2) Why do I get this interesting saw pattern (i dont mean the black random one), so almost everything is found within a couple of percent of screening, then a while nothing, are then somehow some "useful features" adopted allowing me to find a couple of articles (probably belonging to a similar grouping) and then that pattern repeats twice more? The reason I ask is coz the first phase if you let it play out and not cut at 30% seems much smoother. 

Thank you for your insights.

Kind regards,

Rinus Verdonschot
Maastricht University Libray

**Comment by Rensvandeschoot** [comment, marked as answer]
Hi @Rinus-Verdonschot-UB, this is truly a nice simulation study and really helps to understand how the [SAFE procedure](https://doi.org/10.1186/s13643-024-02502-7) with [model switching](https://doi.org/10.3389/frma.2023.1178181) can be used to avoid stopping too early!! Many relevant records outside your screening window in the first simulation run would not have been seen by a human (if you had used 50 consecutive irrelevant records in a row as a stopping rule). 

In the second run, you have used all labels from the first run up to 1164, and the most relevant records not identified in the first run are now detected really quickly! This is a great result!

However, there are still three relevant papers that would not be found if you had used the same stopping rule. These papers are what we call "hard-to-find-relevant papers", see also the simulation study by [Byrne et al.](https://doi.org/10.1186/s13643-024-02587-0). In [Harmsen et al.](https://doi.org/10.1186/s13643-024-02590-5), we also identified a few of such records, and we went back to the initial screeners to ask for possible reasons for these hard-to-find papers. so, I would advise you to: (1) based on Byrne et al, to test if always the same three papers are hard-to-find (e.e., were they also the last three papers in the first simulation run?), and (2) in line with Harmsen et al., to try to explain why these abstracts are so tricky to identity by the classifiers; maybe there is a theoretical explanation, or does the abstract contain no information about the inclusion criteria, etc.

Hope this helps!

**Reply by Rinus-Verdonschot-UB** [reply]
Dear @Rensvandeschoot ,

Thank you this definitely makes sense. I'll read the papers you recommended. Its indeed a good idea to see if its always the same papers which are hard to find!

Kind regards,

Rinus
